claim_batch/services.py

```diff
@@ -431,6 +432,109 @@ def _get_capitation_region_and_district(location_id):
+                .filter(validity_to__isnull = True)\
+                .filter(policy__product = product)
+
+            # 2 batchRun pre-calculation 
+            # 2.1 filter a calculation valid for batchRun with context BatchPrepare (got via 0.2): like allocated contribution
+            if work_data.paymentplans:
+                for paymentplan in work_data.paymentplans:
+                    if paymentplan.calculation.active_for_object(batch_run, 'BatchPrepare'):
+                        # work_data gets updated with relevant data such as work_data.allocated_contribution
+                        paymentplan.calculation.calculate(work_data, 'BatchPrepare'))
+                    # 3 Generate Batchvaluation:
+                    # 3.1 filter a calculation valid for batchRun with context BatchValuation (got via 0.2)
+                    if paymentplan.calculation.active_for_object(batch_run, 'BatchValuate'):
+                    # 3.2 Execute the calculation per Item or service (not claims)
+                        paymentplan.calculation.calculate(work_data, 'BatchValuate'))
+            else:
+                logger.info("no payment plan product found for product  %s for %s/%s", product_id, period, year)
+            # 4 update the claim Total amounts if all Item and services got "valuated"
+            claims = Claim.objects\
+                .filter(validity_from__lte = start_date)\
+                .filter(validity_from__gte = end_date)\
+                .filter(validity_to__isnull = True)\
+                .filter(claim_pocess_stamp_lte = end_date  )\
+                .filter(claim_pocess_stamp_gte = end_date  )\
+                .filter(Q(items__product_in = products) & Q(items__legacy_id__isnull = True) | Q(services__product_in = products) & Q(services__legacy_id__isnull = True))\
+                .prefetch_related(Prefetch('items', queryset=ClaimItem.objects.filter(legacy_id__isnull=True)))\
+                .prefetch_related(Prefetch('services', queryset=ClaimService.objects.filter(legacy_id__isnull=True)))
+            for claim in claims:
+                remunerated_amount = 0
+                for service in claim.services:
+                    remunerated_amount += service.remunerated_amount
+                for item in claim.items:
+                    remunerated_amount += item.remunerated_amount
+            # 5 Generate BatchPayment per product (Ideally per pool but the notion doesn't exist yet)
+            # 5.1 filter a calculation valid for batchRun with context BatchPayment (got via 0.2)
+            if work_data.paymentplans:
+                for paymentplan in work_data.paymentplans:
+                    if paymentplan.calculation.active_for_object(batch_run, 'BatchPayment'):
+                        # 54.2 Execute the converter per product/batch run/claim (not claims)
+                        paymentplan.calculation.calculate(work_data, 'BatchPayment'))
```

Comment:
@delcroip I think we miss the part with creating BatchRun object after execution of batch run. 
like 
```
    from core.utils import TimeUtils
    created_run = BatchRun.objects.create(location_id=location_id, run_year=year, run_month=period,
                                          run_date=TimeUtils.now(), audit_user_id=audit_user_id,
                                          validity_from=TimeUtils.now())
    logger.debug("do_process_batch created run: %s", created_run.id)
```